core/src/apresolve.rs

```diff
@@ -43,27 +43,29 @@ async fn try_apresolve(
     let body = hyper::body::to_bytes(response.into_body()).await?;
     let data: ApResolveData = serde_json::from_slice(body.as_ref())?;
 
+    let mut aps = data.ap_list.into_iter().filter_map(|ap| {
+        let mut split = ap.rsplitn(2, ':');
+        let port = split
+            .next()
+            .expect("rsplitn should not return empty iterator");
+        let host = split.next()?.to_owned();
+        let port: u16 = port.parse().ok()?;
+        Some((host, port))
+    });
```

Comment:
@roderickvd Note that I split up host:port already in apresolve. It just seemed easier for the moment, it's not meant to be final anyway.

Reply:
Seems like a good idea.